### DISCUSSION TITLE
New translation process for website

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1769] [Documentation]
The current translation and release process of the website has been previously discussed in https://github.com/jamulussoftware/jamuluswebsite/issues/14 and https://github.com/jamulussoftware/jamuluswebsite/issues/31 as the project develops, many more people got involved.

The current release process seems to be just to complicated and neither translator nor maintainer friendly. With the upcoming po4a work by ignotus, it's time to rethink the whole process. 

This discussion should be the start of a new translation process and analysis what works and what doesn't.

***

A small technical explanation which might help you understand the website process in a deeper way:

1. We use GitHub pages for serving jamulus.io. This doesn't allow us to use a dynamic website.
2. We use Jekyll, a static site generator which basically takes markdown files, processes them and later on converts them to HTML. This makes translation and reviewing content possible but also involves git which can be complicated. 
3. By using Jekyll, we need a machine on which markdown is converted to HTML. This is currently done by a GitHub action.
3. This GitHub action currently takes everything on the release branch, converts it to html and pushes the html output to a special branch called gh-pages. The html on this branch gets published directly on jamulus.io

To enable translation and changes which don't go live yet, we must use at least one different branch to hold back all the work from getting published. Currently this is either the changes or translate* branch.

I will not explain the process further to enable an open, un-biased discussion.

**Comment by ignotus666** [comment]
Linking to [this discussion](https://github.com/jamulussoftware/jamulus/discussions/1725) that gives a rough description of what the process with po4a might look like.

**Comment by ann0see** [comment]
Ok. Suggestion: We remove the translate branch, freeze changes before a release gets out and then later on merge changes to release. We might even rename changes to next-release and have a similar process to the app:

Once the app translations start, the english version of the website is frozen and the translations should start (preferably with po4a). Once everything is finished, we do a final check locally (every translator should do this for his/her own work). Once we're happy with it, we merge changes (or next-release to release)

**Reply by ignotus666** [reply]
This is one of the main advantages I hope po4a will bring - all of us being able to work on a single branch where all changes happen. First modifications to EN files, freeze, then translations, and finally an overall check before merging to release. No confusions as to where to merge changes and no headaches for translators pulling their hair out trying to track down those changes in different branches.

After the upcoming release I propose doing some tests with po4a to get the feel of it, its possibilities, and seeing whether the above works. I'm fairly confident it should, but unforeseen stuff always crops up...
